### PR TITLE
Check MySQL version in spree_credit_cards data json field

### PR DESCRIPTION
**Pull request by dfranciosi**
If not supported downgrade to a text field

### CODE REVIEW

**Comment by kamui** [comment]
Assigning to @allisonlarson because she worked directly on this code and this is something we tried to support initially. My vague recollection is that if the type is `:json`, activerecord will auto marshall/unmarshall when you access the attribute, if it's text it's something you have to do manually.

File: db/migrate/20150910170527_add_data_to_credit_card.rb [hunk 1]
```diff
@@ -1,5 +1,9 @@
 class AddDataToCreditCard < ActiveRecord::Migration
   def change
-    add_column :spree_credit_cards, :data, :json
+    if ActiveRecord::Base.connection.adapter_name != 'SQLite' && ActiveRecord::Base.connection.supports_json?
```
Comment (dfranciosi) [inline db/migrate/20150910170527_add_data_to_credit_card.rb]: @kamui @allisonlarson Please pay attention that ``` supports_json? ``` has been merged here:
https://github.com/rails/rails/pull/21110
but it's still not part of the latest minor of ```active_record (4.2.5) ```.
It's a better version but still an edge one :smiley:

Comment (athal7) [inline db/migrate/20150910170527_add_data_to_credit_card.rb]: @dfranciosi We should make sure to update the rails dependency then, and to cut a new version of the gem.

**Comment by dfranciosi** [comment]
@allisonlarson any news? :wink:

**Comment by dfranciosi** [comment]
@allisonlarson do you have a feedback for me? thanks in advance /cc @kamui

**Comment by athal7** [comment]
@dfranciosi I don't think you have addressed my concern

**Comment by allisonlarson** [comment]
@dfranciosi Sorry for the late responses, holidays are a busy time. I am going to agree with @athal7 for the ActiveRecord version. 

Also, my recollection of `:text` vs `:json` follows @kamui. So i'm not sure if just changing the field type would work, since the `:text` field would require to manually marshal/unmarshall.

File: db/migrate/20150910170527_add_data_to_credit_card.rb (hunk 1 above)
Comment (dfranciosi) [inline db/migrate/20150910170527_add_data_to_credit_card.rb]: @athal7 looking at the updates of that ```rails``` PR, their idea is to include ```supports_json?``` methods only in rails 5.0 so my suggestion is to remove my second commit to be compliant to the current version of ```ActiveRecord```, what do you think? /cc @kamui

**Comment by jhawthorn** [comment]
This should be solved by #16, which just always uses a text field.

In the not-too-distant future I'd like to see us use a proper `PaypalAccount` source (or similar) to hold this information.

**Comment by dfranciosi** [comment]
@jhawthorn 👍

**Pull request closed without merging**